### PR TITLE
Python: be explicit on whether it is python2 or python3 that is used

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -14131,8 +14131,6 @@ in
     autoreconfHook = buildPackages.autoreconfHook269;
   };
 
-  gdal_1_11 = callPackage ../development/libraries/gdal/gdal-1_11.nix { };
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix:14134]: https://github.com/NixOS/nixpkgs/pull/117587

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117593` at 538d09b1 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages skipped due to time constraints:</summary>
  <ul>
    <li>gede</li>
    <li>gen-oath-safe</li>
    <li>gitstats</li>
    <li>gnss-sdr</li>
    <li>gnubg</li>
    <li>gnuradio3_8Packages.limesdr</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gnuradio3_8Packages.rds</li>
    <li>gtimelog</li>
    <li>qradiolink</li>
  </ul>
</details>
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>ginac</li>
    <li>global</li>
    <li>gnuradio3_7Packages.python</li>
    <li>gnuradioPackages.python (gnuradio3_8Packages.python)</li>
    <li>grabserial</li>
    <li>gup</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/development/tools/build-managers/gup/default.nix:16:3:
```
   |
16 |   buildPhase = "make python";
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/build-managers/gup/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/science/math/ginac/default.nix:27:5:
```
   |
27 |     license = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/os-specific/linux/libnl/default.nix [hunk 2]
```diff
@@ -1,5 +1,5 @@
 { stdenv, file, lib, fetchFromGitHub, autoreconfHook, bison, flex, pkg-config
-, pythonSupport ? false, swig ? null, python}:
+, pythonSupport ? false, swig ? null, python ? null}:
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libnl/default.nix:2]: Why should swig or python be null?

File: pkgs/games/gnubg/default.nix [hunk 3]
```diff
@@ -9,8 +9,10 @@ stdenv.mkDerivation {
     sha256 = "11xwhcli1h12k6rnhhyq4jphzrhfik7i8ah3k32pqw803460n6yf";
   };
 
-  nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ python glib gtk2 readline ];
+  nativeBuildInputs = [ pkg-config python3 glib ];
+  buildInputs = [  gtk2 readline ];
```
Comment (SuperSandro2000) [inline pkgs/games/gnubg/default.nix:13]:
```suggestion
  buildInputs = [ gtk2 readline ];
```

File: pkgs/development/tools/build-managers/gup/default.nix [hunk 4]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, nix-update-source, lib, python
+{ stdenv, fetchFromGitHub, nix-update-source, lib, python3
 , which, runtimeShell, pychecker ? null }:
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/gup/default.nix:2]:
```suggestion
, which, runtimeShell, pychecker }:
```

File: pkgs/development/libraries/grib-api/default.nix [hunk 5]
```diff
@@ -1,6 +1,6 @@
 { fetchurl, fetchpatch, lib, stdenv,
   cmake, netcdf, gfortran, libpng, openjpeg,
-  enablePython ? false, pythonPackages }:
+  enablePython ? false, pythonPackages ? null }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/grib-api/default.nix:3]: Why should pythonPackages be null?

File: pkgs/applications/audio/gtklick/default.nix [hunk 6]
```diff
@@ -1,6 +1,6 @@
-{ lib, fetchurl, pythonPackages, gettext, klick}:
+{ lib, fetchurl, python2Packages, gettext, klick}:
 
-pythonPackages.buildPythonApplication rec {
+python2Packages.buildPythonApplication rec {
```
[the remark this answers is not in the document]
Comment (wamserma) [reply pkgs/applications/audio/gtklick/default.nix:3]: hmm.. I gave it a try. Builds, but crashes on run: https://github.com/wamserma/nixpkgs/tree/gtklick-py3

File: pkgs/applications/misc/kiwix/default.nix [hunk 7]
```diff
@@ -105,5 +105,6 @@ stdenv.mkDerivation rec {
     knownVulnerabilities = [
       "CVE-2015-1032"
     ];
+    broken = true; # marked broken at 2020-03-26
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/kiwix/default.nix:108]: Why is it marked broken?

File: pkgs/top-level/all-packages.nix [hunk 8]
```diff
@@ -5862,8 +5862,6 @@ in
 
   kdbg = libsForQt5.callPackage ../development/tools/misc/kdbg { };
 
-  kippo = callPackage ../servers/kippo { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:5865]: We should add an alias for the package.

File: pkgs/development/tools/misc/kconfig-frontends/default.nix [hunk 9]
```diff
@@ -10,17 +10,15 @@ stdenv.mkDerivation rec {
     url = "http://ymorin.is-a-geek.org/download/${basename}/${name}.tar.xz";
   };
 
-  nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ bison flex gperf ncurses pythonPackages.python pythonPackages.wrapPython ];
+  nativeBuildInputs = [ bison flex gperf pkg-config ];
+  buildInputs = [ bash ncurses python3 ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/kconfig-frontends/default.nix:14]: Bash as an buildInputs? That seems very strange.

File: pkgs/development/libraries/kmsxx/default.nix [hunk 10]
```diff
@@ -16,13 +16,14 @@ stdenv.mkDerivation {
...
-  buildInputs = [ libdrm python ];
+  buildInputs = [ libdrm ] ++ lib.optionals withPython [ python ];
 
   meta = with lib; {
     description = "C++11 library, utilities and python bindings for Linux kernel mode setting";
     homepage = "https://github.com/tomba/kmsxx";
     license = licenses.mpl20;
     maintainers = with maintainers; [ gnidorah ];
     platforms = platforms.linux;
+    broken = true; # marked broken 2021-03-26
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kmsxx/default.nix:27]: Why got it marked broken?

File: pkgs/applications/misc/kiwix/default.nix (hunk 7 above)
Comment (ajs124) [reply pkgs/applications/misc/kiwix/default.nix:108]: Because it's broken right now. Fix in #117851

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (FRidh) [reply pkgs/top-level/all-packages.nix:14269]: added alias

**Pull request merged**